redis/tests/test_async.rs

```diff
@@ -781,3 +782,52 @@ mod mtls_test {
         }
     }
 }
+
+#[test]
+fn test_set_client_name_by_config() {
+    const CLIENT_NAME: &str = "TEST_CLIENT_NAME";
+    use redis::RedisError;
+    let ctx = TestContext::with_client_name(CLIENT_NAME);
+
+    block_on_all(async move {
+        let mut con = ctx.async_connection().await?;
+
+        let client_list: redis::InfoDict = redis::cmd("CLIENT")
+            .arg("LIST")
+            .query_async(&mut con)
+            .await
+            .unwrap();
+        assert!(
+            !client_list.is_empty(),
+            "No clients present in CLIENT LIST result"
+        );
+
+        let val: &redis::Value = client_list.values().next().unwrap();
+
+        match val {
+            redis::Value::SimpleString(s) => {
+                let client_attr: Vec<&str> = s.split(' ').collect();
+                for i in client_attr.iter() {
+                    let this_attr: Vec<&str> = i.split('=').collect();
+                    if this_attr[0] == "name" {
+                        assert!(
+                            this_attr[1] == CLIENT_NAME,
```

Comment:
You can take an example from this test:
https://github.com/amazon-contributing/redis-rs/pull/56/files#diff-85259ca2185589cf8979102fc91c6ce1a9461aac8ed605472b9ad53c3324c400R1912

Reply:
Summarizing our talk yesterday - the test will be changed to check the client ID and based on the it would check that its name is the one passed in the config